### ISSUE TITLE
feat(rules): enforce Microdata cross-attribute constraints (itemid/itemtype require itemscope)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3733]
## Summary

Enforce the Microdata cross-attribute presence constraint: `itemid` requires both `itemscope` and `itemtype` to be present on the same element.

## Background

Split from #3597 (closed). The original issue claimed markuplint doesn't validate `itemtype` / `itemid` URL values, but that is already implemented by `invalid-attr`. The only genuine gap is the cross-attribute constraint.

## What to enforce

```html
<!-- Invalid: itemid without itemscope/itemtype -->
<div itemid="https://example.com/resource"></div>

<!-- Invalid: itemid with itemscope but no itemtype -->
<div itemscope itemid="https://example.com/resource"></div>

<!-- Valid -->
<div itemscope itemtype="https://schema.org/Thing" itemid="https://example.com/resource"></div>
```

## Spec reference

https://html.spec.whatwg.org/multipage/microdata.html#attr-itemid

> The itemid attribute must not be specified on elements that do not have both an itemscope attribute and an itemtype attribute specified

## Verification

Run nu-html-checker via Docker to verify real output before implementation:
```bash
docker run --rm -i ghcr.io/validator/validator:latest vnu --format json - < test.html
```

## Affected package

`@markuplint/rules` — new rule or extension of existing attribute presence check

## Benchmark cross-reference

_Sync'd from `tests/external/snapshots/` (submodule=1429313, nu-validator@sha256:59b0e97e2664, markuplint=5.0.0-rc.4, generated=2026-04-27 06:10:57 UTC)._

Filter: `itemid-without-itemtype|itemid-without-itemscope|itemtype-without-itemscope` → **6** fixtures. Verdict tally: match-error=0, match-clean=0, ml-only=0, nu-only=6, nu-over=0.

All six fixtures are `nu-only` — markuplint misses itemid/itemtype cross-attribute constraints. HTML LS §5.7 makes `itemid` require both `itemscope` and `itemtype`.

| fixture | verdict | nu id / ml rule |
| --- | --- | --- |
| `html/assertions/itemid-without-itemtype-novalid.html` | nu-only | `nv-1db1d5638ec2` |
| `html/assertions/itemtype-without-itemscope-novalid.html` | nu-only | `nv-111aeeae2559` |
| `html/elements/microdata/itemid-without-itemtype-novalid.html` | nu-only | `nv-ab8ae7adfffb` |
| `html/microdata/itemid-without-itemscope-novalid.html` | nu-only | `nv-fb4907b5f807` |
| `html/microdata/itemid-without-itemtype-novalid.html` | nu-only | `nv-5aa9875cf757` |
| `html/microdata/itemtype-without-itemscope-novalid.html` | nu-only | `nv-e1040f24a999` |

Reproduce: see [`tests/external/CLAUDE.md`](https://github.com/markuplint/markuplint/blob/dev/tests/external/CLAUDE.md) "Auditing a claim against the benchmark".

**Comment by YusukeHirao** [comment]
## Docker 実測結果（2026-04-15）

### 全 4 ケース対比

| # | HTML | nu-html-checker | markuplint | Gap |
|---|------|----------------|-----------|-----|
| 1 | `<div itemid="https://example.com/r1">` | `The "itemid" attribute must not be specified on elements that do not have both an "itemscope" attribute and an "itemtype" attribute specified.` | （検出なし） | **YES** |
| 2 | `<div itemscope itemid="https://example.com/r2">` | 同上 | （検出なし） | **YES** |
| 3 | `<div itemscope itemtype="..." itemid="...">` (正常) | 違反なし | 違反なし | NO |
| 4 | `<div itemtype="..." itemid="...">` (itemscope なし) | `itemid...` + `The "itemtype" attribute must not be specified on elements that do not have an "itemscope" attribute specified.` | （検出なし） | **YES**（2 種類の制約） |

## スコープ拡張: 2 つの制約

本 Issue は `itemid` のみを対象としていたが、nu は **`itemtype` の独立制約** も報告する。まとめて実装すべき:

1. **`itemid` requires `itemscope` AND `itemtype`** on the same element
2. **`itemtype` requires `itemscope`** on the same element

## 実装方針

`@markuplint/rules` に cross-attribute 制約ルール追加（または既存 `required-attr` 拡張）。

推奨メッセージ形式（nu 準拠）:
- `The "itemid" attribute must not be specified on elements that do not have both an "itemscope" attribute and an "itemtype" attribute specified.`
- `The "itemtype" attribute must not be specified on elements that do not have an "itemscope" attribute specified.`
